### PR TITLE
Prevent throwing NoSuchMethodException on direct getMethod calls

### PR DESCRIPTION
**Pull request by oskar-szwajkowski**
Instead, get list of methods and filter it

Such exception are expensive in terms of generating stack, plus they unnecessarily bloat JVM reports with tons of throws

### CODE REVIEW

**Comment by oskar-szwajkowski** [comment]
@martint @dain  Hey, could any of you you look into this and could we merge this?

**Review by wendigo — APPROVED** [review]
LGTM

File: src/main/java/org/weakref/jmx/AnnotationUtils.java [hunk 1]
```diff
@@ -336,4 +333,15 @@ private static boolean isAnnotationPresent(Class<? extends Annotation> annotatio
 
         return false;
     }
+
+    private static Optional<String> invokeSwallowingException(Method method, Annotation annotation)
```
Comment (martint) [inline src/main/java/org/weakref/jmx/AnnotationUtils.java:337]: Call this `tryInvoke`

Comment (oskar-szwajkowski) [reply src/main/java/org/weakref/jmx/AnnotationUtils.java]: done

**Pull request merged**